magento/pwa-studio
# Header Search Dropdowns

## Description

This PR updates the Search trigger button in the site header to behave like my account and cart trigger buttons for consistency.
This also solves the issue of two dialogs overlapping each other.

It also:
  * Removes `searchOpen` and its corresponding `toggleSearch` action from global app state (Redux). The search trigger controls this piece of state entirely now.
  * Treats the search input (`SearchBar` component) as a `dropdown`, triggered by `SearchTrigger`.
  * Preserves search input behavior such as focusing it automatically when search is opened
  * No longer turns the Search Trigger text blue when the search dropdown is open (consistency).
  * Adds a "Showing results for {search term}" message on the search page

## Related Issue





Closes PWA-786.

## Acceptance 



### Verification Stakeholders


@schensley 

### Specification



### Verification Steps



1. Verify that you cannot have both the Search and My Account (or MiniCart) triggers open at the same time

#### No Loss of Functionality

For this section, compare this PR deployment with https://develop.pwa-venia.com/.

1. Verify that opening search works the same
1. Verify that closing search works the same except that now, clicking outside of the search area or the search trigger will close search as well
1. Verify that autocomplete works
1. Verify that you can execute a search

On the Search Results page (`/search.html`):

1. Verify that opening the search text field pre-populates it with the last executed search term (if any)
1. Verify that the "Showing results for..." message is accurate, both with and without a category (get a category by clicking a suggested category in the autocomplete dropdown)

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
